jujugui/static/gui/src/app/app.js

```diff
@@ -2585,6 +2591,18 @@ YUI.add('juju-gui', function(Y) {
           );
         }, this);
       }
+      // Retrieve maas credentials should they exist.
```

Comment:
IIUC this onLogin in the app is called after the user logs into the model. What we really want I think is running this after the controller connection is authenticated. I'd suggest to move this code to controller-api.js in the handleLogin method, if the auth succeeds.